lua/telescope/utils.lua

```diff
@@ -170,6 +170,113 @@ utils.filter_symbols = function(results, opts, post_filter)
   end
 end
 
+utils.is_path_hidden = function(opts, path_display)
+  path_display = path_display or vim.F.if_nil(opts.path_display, require("telescope.config").values.path_display)
+
+  return path_display == nil
+    or path_display == "hidden"
+    or type(path_display) == "table" and (vim.tbl_contains(path_display, "hidden") or path_display.hidden)
+end
+
+utils.is_uri = function(filename)
+  local char = string.byte(filename, 1) or 0
+
+  -- is alpha?
+  if char < 65 or (char > 90 and char < 97) or char > 122 then
+    return false
+  end
+
+  for i = 2, #filename do
+    char = string.byte(filename, i)
+    if char == 58 then -- `:`
+      return i < #filename and string.byte(filename, i + 1) ~= 92 -- `\`
+    elseif
+      not (
+        (char >= 48 and char <= 57) -- 0-9
+        or (char >= 65 and char <= 90) -- A-Z
+        or (char >= 97 and char <= 122) -- a-z
+        or char == 43 -- `+`
+        or char == 46 -- `.`
+        or char == 45 -- `-`
+      )
+    then
+      return false
+    end
+  end
+  return false
+end
+
+local calc_result_length = function(truncate_len)
+  local status = get_status(vim.api.nvim_get_current_buf())
+  local len = vim.api.nvim_win_get_width(status.layout.results.winid) - status.picker.selection_caret:len() - 2
+  return type(truncate_len) == "number" and len - truncate_len or len
+end
+
+utils.path_filename_first = function(path, path_display)
```

Comment:
lets actually do some refactoring and lets not take in the whole `path_display` tbl but only `path_display["filename_first"]` that makes it clearer that the function only operates on that argument

Reply:
I agree. I would be in favor of just passing the primitives instead of a potential table. It makes (future) function composition easier, as you don't have to construct artificial objects.

What do you say?

Reply:
> primitives

not really primitives but yeah

> What do you say

yeah, thats why i requested the change thanks